Fullscreen loader logging

kidroca opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->
As per the ticket requirement we're adding logging to Fullscreen loaders in order to capture a loader never being removed from the screen
When `name` and `timeout` props are used on the loader, it would log a message when it mounts/unmounts or stays longer than the specified timeout
In order to have simpler component did mount / will unmount logic, the `visible` prop was removed and existing usages refactored. `FullScreenLoadingIndicator` is used uniformly across the app by conditional rendering: 
```jsx
{this.state.isLoading && <FullScreenLoadingIndicator />}
```


### Fixed Issues
<!---
Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing.
Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the issue; otherwise, the linking will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<number-of-the-issue>

Do NOT only link the issue number like this: $ #<number-of-the-issue>
--->
$ https://github.com/Expensify/App/issues/7424

### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- [ ] Verify `[LoadingIndicator] "${this.props.name}" became visible` is logged when loader is visible
    - logged for NavigationRoot when launching App (on ios/android)
    - logged for MainDrawerNavigation after login (on all platforms)
- [ ] Verify `[LoadingIndicator] "${this.props.name}" disappeared` is logged when loader hides
     - logged for NavigationRoot  when launching App (on ios/android)
     - logged for MainDrawerNavigation after login (on all platforms)
- [ ] Verify `[LoadingIndicator] "${this.props.name}" is still visible after ${this.props.timeout} ms` is logged when loader is displayed for longer than `timeout` duration
- [ ] There should be no regressions on places that use the `FullscreenLoadingIndicator`

I was able to test the `timeout` message (3rd bullet) when I forced the MainDrawerLoader to never hide. It also happened once (on dev) because loading was too slow and took longer than 15 sec. (after signing in)

#### Sample logging output
![image](https://user-images.githubusercontent.com/12156624/160834930-8c177d39-9a2d-4514-939c-2e345a4a0fe7.png)
note: _Since "Disappeared" is logged on unmount, the component didn't received the new props and logs whatever `logDetail` it last had, hence `reportID` is still `""` in the log_

### PR Review Checklist
<!--
This is a checklist for PR authors & reviewers. Please make sure to complete all tasks and check them off once you do, or else Expensify has the right not to merge your PR!
-->
#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.


<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

No feature or bug to be tested. 
We're testing against any regressions around fullscreen loading indicators. 
A fullscreen loading indicator is a small rotating circle in the middle of the page with semi-transparent (white/gray) background

On all platforms, verify loading circle still appears:
- [ ] briefly after sign in
- [ ] on Wallet statement modal, while statement is loading
- [ ] after selecting "New chat" and search/users load
- [ ] after selecting "Search" and search/users load
- [ ] when you select a chat in LHN and it's loaded in the main view
- [ ] on the IOU modal while it loads
- [ ] on the Workspace / Manage Members / Invite page, while search/users load

_(These are the pages that had the `visible` prop which was removed from the loader)_

### Screenshots
<!-- Add screenshots for all platforms tested. Pull requests won't be merged unless the screenshots show the app was tested on all platforms.-->

#### Web
<!-- Insert screenshots of your changes on the web platform-->
![image](https://user-images.githubusercontent.com/12156624/160684233-40219b63-85e7-4e0f-ba0a-69c83d75d64f.png)

#### Mobile Web
<!-- Insert screenshots of your changes on the web platform (from a mobile browser)-->
![image](https://user-images.githubusercontent.com/12156624/160686914-feb8d211-cf86-4c77-9de2-99bdb82879d5.png)

#### Desktop
<!-- Insert screenshots of your changes on the desktop platform-->
![image](https://user-images.githubusercontent.com/12156624/160684487-389c71c5-3d94-454b-b2d2-ad9ad72dfdae.png)

#### iOS
<!-- Insert screenshots of your changes on the iOS platform-->
![image](https://user-images.githubusercontent.com/12156624/160685380-7d224510-ec9a-4e89-aa71-47dd70d5b145.png)

#### Android
<!-- Insert screenshots of your changes on the Android platform-->
https://user-images.githubusercontent.com/12156624/160691993-8224b64a-2661-4a21-9517-b90d5a9ed43b.mp4

kidroca committed:
Add logging to fullscreen loaders to investigate infinite loading screens

kidroca committed:
FullscreenLoadingIndicator - don't print additional noise from stacktrace

kidroca committed:
Add loader timeout logs for MainDrawerNavigator and NavigationRoot

marcaaron reviewed (changes requested):
Can we pause reviews on this pending answers to the questions in this comment? Thanks 🙇 

https://github.com/Expensify/App/issues/7424#issuecomment-1082334819

marcaaron commented on src/components/FullscreenLoadingIndicator.js:
```diff
@@ -5,17 +5,22 @@ import {ActivityIndicator, StyleSheet, View} from 'react-native';
+    name: PropTypes.string,
+
+    /** Optional duration (ms) after which a message would be logged if the loader is still covering the screen */
+    timeout: PropTypes.number,
```
**suggestion:** Rather than have a custom timeout here we can just make this `shouldLogAlertOnTimeout` since we use 15 seconds everywhere anyway and there are not cases where we need a more specific timeout (and maybe never will be)

kidroca replied:
In order to capture meaningful logs we need to have `logDetail` with at least the screen name, or some name we can use to distinct log messages

Instead of introducing a flag like `shouldLogAlertOnTimeout` we use the optional `logDetail` prop - if it is provided it's an indication that we'd like to capture logs

If you prefer to have an explicit flag anyway let me know and I'll update

marcaaron commented on src/components/FullscreenLoadingIndicator.js:
```diff
@@ -5,17 +5,22 @@ import {ActivityIndicator, StyleSheet, View} from 'react-native';
-    /** Controls whether the loader is mounted and displayed */
-    visible: PropTypes.bool,
+    /** Name used in the logs if the loader is displayed for too long time */
+    name: PropTypes.string,
```
**suggestion:** `logDetail` or `logContent` to make it clear what the `name` will be used for

marcaaron commented on src/components/FullscreenLoadingIndicator.js:
```diff
@@ -25,18 +30,42 @@ const defaultProps = {
-        return null;
+class FullScreenLoadingIndicator extends React.Component {
+    componentDidMount() {
+        if (!this.props.name || !this.props.timeout) {
```
**suggestion:** We could still log without additional context. But if we want to enforce additional context then maybe we should throw and return so that someone implementing this will remember to add details.

kidroca replied:
Good point in order to distinct the loader, we should require some context - at least the screen name where it's used, so we should throw an error - if someone tries to setup a new or existing loader with logging incorrectly they'll know right away

kidroca replied:
Updated the logic to throw an error if `props.logDetail` exists but `logDetail.name` is not set

But since `logDetail.name` is flagged as `PropTypes.required` I don't know whether we need to throw an exception, and maybe we should remove it and just return

marcaaron commented on src/components/FullscreenLoadingIndicator.js:
```diff
@@ -25,18 +30,42 @@ const defaultProps = {
+            return;
+        }
+
+        Log.info(`[LoadingIndicator] "${this.props.name}" became visible`);
```
**suggestion:** Change these up so that the additional log detail is passed as a parameter instead of a string. And just have `Log.info('[LoadingIndicator] Became visible', {additionalLogDetail})` or whatever we want to name that param. 

**opinion based on log search preferences:** Can be slightly simpler to grep without a * pattern and easier to find logs when the content does not change very often.

marcaaron commented on src/components/FullscreenLoadingIndicator.js:
```diff
@@ -25,18 +30,42 @@ const defaultProps = {
+
+        Log.info(`[LoadingIndicator] "${this.props.name}" became visible`);
+
+        this.timeoutId = setTimeout(
```
```suggestion
        this.timeoutID = setTimeout(
```

marcaaron commented on src/components/WalletStatementModal/index.js:
```diff
@@ -41,9 +41,7 @@ class WalletStatementModal extends React.Component {
-                <FullScreenLoadingIndicator
-                    visible={this.state.isLoading}
-                />
+                {this.state.isLoading && <FullScreenLoadingIndicator />}
```
**thought (not blocking):** Looks like the video where we saw the long spinner was after a login so it's probably not this one.

marcaaron replied:
There are some others here where we can probably assume the same. But maybe we should just always log for all slow spinners with or without additional context.

marcaaron replied:
Ah ok I see now. And it is easier to remove the prop and track mount and unmount vs. changes to `props.isLoading`

marcaaron replied:
Nice change 👍

marcaaron commented on src/libs/Navigation/AppNavigator/MainDrawerNavigator.js:
```diff
@@ -49,7 +49,7 @@ const MainDrawerNavigator = (props) => {
     // Wait until reports are fetched and there is a reportID in initialParams
     if (!initialParams.reportID) {
-        return <FullScreenLoadingIndicator />;
+        return <FullScreenLoadingIndicator name="Main Drawer Loader" timeout={15 * 1000} />;
```
**suggestion:** Move the `15 * 1000` into a `CONST.TIMING.SPINNER_TIMEOUT` variable

marcaaron commented on src/components/FullscreenLoadingIndicator.js:
```diff
@@ -25,18 +30,42 @@ const defaultProps = {
+
+        this.timeoutId = setTimeout(
+            () => Log.alert(
+                `[LoadingIndicator] "${this.props.name}" is still visible after ${this.props.timeout} ms`,
```
It's impossible to know this. But our alert logs must hit a threshold in order to actually create a ticket for us to investigate. I feel like with the low numbers of users we should use it here. There's a constant we can add and I'd change this to:

```js
Log.alert(`${CONST.ERROR.ENSURE_BUGBOT} [LoadingIndicator] Visible after timeout`,  details, false);
```

and put the timeout length and whatever other details into params

marcaaron reviewed (changes requested):
Overall, I like this now that I understand the changes. Did have some suggestions though.

kidroca reviewed (commented):
Left some notes, will address the changes tomorrow

BTW 
I haven't checked some PR check items, because I'm not certain what I should do.
Waiting to be clarified in this slack thread: https://expensify.slack.com/archives/C01GTK53T8Q/p1648583058174529

kidroca committed:
Update FullScreenLoadingIndicator with PR suggestions

- remove `name` and `timeout` props. Use fixed timeout and a `logDetail` prop
- use fixed strings in log messages and pass details as additional data
- extract timeout duration to CONST
- throw if logging is not configured correctly
- trigger a special CONST.ERROR.ENSURE_BUGBOT alert

kidroca reviewed (commented):
Added changes from the provided suggestions

The updated logging looks like this:
![image](https://user-images.githubusercontent.com/12156624/160834930-8c177d39-9a2d-4514-939c-2e345a4a0fe7.png)
- note: since "Disappeared" is logged on unmount, the component didn't received the new props and logs whatever `logDetail` it last had, hence `reportID` is still `""` in the log

A new change is the `Timing` API 
We capture the duration for which a loader was visible using `Timing.start/end` 
If that's undesired we can just discard the last commit

parasharrajat reviewed (commented):
Looking good.

kidroca commented:
@marcaaron PR is ready for review

thienlnam reviewed (dismissed):
Changes look good to me 👌

parasharrajat commented on src/components/FullscreenLoadingIndicator.js:
```diff
@@ -25,18 +34,48 @@ const defaultProps = {
  * @param {Object} props
  * @returns {JSX.Element}
  */
```
```suggestion
```

parasharrajat commented on src/components/FullscreenLoadingIndicator.js:
```diff
@@ -25,18 +34,48 @@ const defaultProps = {
+        }
+
+        if (!this.props.logDetail.name) {
+            throw new Error('A name should be set to distinct logged messages. Please check the `logDetails` prop');
```
```suggestion
            throw new Error('A name should be set to distinct logged messages. Please check the `logDetails` prop.');
```

parasharrajat commented on src/components/FullscreenLoadingIndicator.js:
```diff
@@ -25,18 +34,48 @@ const defaultProps = {
-const FullScreenLoadingIndicator = (props) => {
-    if (!props.visible) {
-        return null;
+class FullScreenLoadingIndicator extends React.Component {
```
Let's add a comment here telling why do we need the class component here so.

kidroca replied:
Other class components do not explain why a class component is used. Why is it any different here?

parasharrajat replied:
Yeah, just checked the style guide again. we are fine here. Point was that this component does not have state or bound methods so it is not apparent that it needs to be a class component. So leaving a comment will prevent others from thinking to change this. But it's fine code says it all. we need lifecycle methods that are clearly visible here.

marcaaron commented:
>A new change is the Timing API We capture the duration for which a loader was visible using Timing.start/end If that's undesired we can just discard the last commit

Let's undo this as it feels like we're just tacking on extra scope. The timing stuff is pretty focused on a few key metrics and we don't even know what our issue is here. Let's not just time things for the sake of timing them.

kidroca committed:
Remove FullscreenLoadingIndicator class description

Other components do not use class description
Removed to be uniform with the rest of our components

kidroca commented:
> > A new change is the Timing API We capture the duration for which a loader was visible using Timing.start/end If that's undesired we can just discard the last commit
> 
> Let's undo this as it feels like we're just tacking on extra scope. The timing stuff is pretty focused on a few key metrics and we don't even know what our issue is here. Let's not just time things for the sake of timing them.

Dropped the commit that introduced this change
Removed class level documentation - other components do not have such

Ready for review

parasharrajat reviewed (approved):
LGTM. 

cc: @thienlnam 
#### PR Reviewer Checklist
- [x] I verified the PR has a small number of commits behind `main`
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the testing environment is mentioned in the test steps
- [x] I verified testing steps cover success & fail scenarios (if applicable)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [ ] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors related to changes in this PR
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified comments were added when the code was not self explanatory
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files (if applicable)
    - [x] I verified proper naming convention for platform-specific files was followed (if applicable)
    - [x] I verified [style guidelines](https://github.com/Expensify/App/blob/main/STYLING.md) were followed
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components are not impacted by changes in this PR (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified the UI performance was not affected (the performance is the same than `main` branch)
- [ ] If a new component is created I verified that a similar component doesn't exist in the codebase
:ribbon: :eyes: :ribbon:  C+ reviewed

thienlnam reviewed (approved):
Changes look good, leaving to @marcaaron

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/2104173077) to staging by @marcaaron in version: 1.1.52-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/2113739670) to production by @roryabraham in version: 1.1.52-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅
